packages/starknet-snap/src/createAccount.ts

```diff
@@ -78,47 +70,8 @@ export async function createAccount(params: ApiParams, silentMode = false, waitM
           };
       }
 
-      const signerAssigned = await isAccountDeployed(network, contractAddress);
-
-      if (!signerAssigned) {
-        try {
-          const balance = await getBalance(
-            getEtherErc20Token(state, network.chainId)?.address,
-            num.toBigInt(contractAddress).toString(10),
-            network,
-          );
-          logger.log(`createAccount:\ngetBalanceResp: ${balance}`);
-          estimateDeployFee = await estimateAccountDeployFee(
-            network,
-            contractAddress,
-            contractCallData,
-            publicKey,
-            privateKey,
-          );
-          logger.log(`createAccount:\nestimateDeployFee: ${toJson(estimateDeployFee)}`);
-          if (Number(balance) < Number(estimateDeployFee.suggestedMaxFee)) {
-            const gasFeeStr = ethers.utils.formatUnits(estimateDeployFee.suggestedMaxFee.toString(10), 18);
-            const gasFeeFloat = parseFloat(gasFeeStr).toFixed(6); // 6 decimal places for ether
-            const gasFeeInEther = Number(gasFeeFloat) === 0 ? '0.000001' : gasFeeFloat;
-            failureReason = `The account address needs to hold at least ${gasFeeInEther} ETH for deploy fee`;
-          }
-        } catch (err) {
-          failureReason = 'The account address ETH balance cannot be read';
-          logger.error(`createAccount: failed to read the ETH balance of ${contractAddress}: ${err}`);
-        }
-      }
-
-      const deployResp = await deployAccount(
-        network,
-        contractAddress,
-        contractCallData,
-        publicKey,
-        privateKey,
-        undefined,
-        {
-          maxFee: estimateDeployFee?.suggestedMaxFee,
-        },
-      );
+      // Deploy account will auto estimate the fee from the network if not provided
+      const deployResp = await deployAccount(network, contractAddress, contractCallData, publicKey, privateKey);
```

Comment:
What happens if createAccount is called on an account that is already deployed ?

Reply:
it will fail in this call, which is the same behaviour as prev logic